corehq/apps/translations/generators.py

```diff
@@ -58,6 +60,13 @@ def __init__(self, domain, app_id, version, key_lang, source_lang, lang_prefix,
         self.slug_to_name[MODULES_AND_FORMS_SHEET_NAME] = {'en': MODULES_AND_FORMS_SHEET_NAME}
         self._build_translations()
 
+    @cached_property
+    def build_id(self):
```

Comment:
where is this used?

Reply:
transifex/parser.py line 35

Reply:
this is `AppTranslationsGenerator` and thats called on `transifex` object?

Reply:
Yeah. In translations.tasks.py line 166 an `AppTranslationsGenerator` instance is assigned to a variable named `gen`, and passed to `TranslationsParser.__init__()`. `TranslationsParser` assigns it to `self.transifex` in transifex/parser.py line 27.

Reply:
Maybe these variables could be named better. :grimacing: I'm open to suggestions.

Reply:
Ya, that expects a `Transifex` object but now we are passing an `AppTranslationsGenerator`?

Reply:
Yes. (@emord wrote this code, and maybe he can clarify better than me, but I'll give it a shot.) `AppTranslationsGenerator` has intentionally been ducktyped so that it can be passed as a `Transifex` instance. This allows the `download_project_from_hq()` task to generate an Excel file without interacting with the Transifex API.

Reply:
Thats confusing, i think that should approach should be reconsidered.

Reply:
```
>>> import this
The Zen of Python, by Tim Peters

...
There should be one-- and preferably only one --obvious way to do it.
Although that way may not be obvious at first unless you're Dutch.
...
```
:joy: 

I'll give it some thought. Another alternative might be classical inheritance, which comes with its own set of baggage. Ducktyping may not be immediately obvious, especially if you're coming from a statically-typed background, but it is a powerful and beautiful thing. Perhaps a compromise would be to rename the `gen` variable to `quacks_like_a_transifex`. :grinning: